### PR TITLE
Fix object-expr untested cases

### PR DESCRIPTION
**Pull request by edgarfgp**
## Description

Continuation of https://github.com/dotnet/fsharp/pull/17388

Found some cases that weren't covered/tested before 

## Checklist

- [x] Test cases added

### CODE REVIEW

**Comment by edgarfgp** [comment]
This is ready.

File: src/Compiler/Checking/Expressions/CheckSequenceExpressions.fs [hunk 1]
```diff
@@ -444,20 +444,12 @@ let TcSequenceExpressionEntry (cenv: TcFileState) env (overallTy: OverallTy) tpe
...
     | Some replacementExpr -> TcExpr cenv overallTy env tpenv replacementExpr
     | None ->
-
         let implicitYieldEnabled =
             cenv.g.langVersion.SupportsFeature LanguageFeature.ImplicitYield
 
         let validateObjectSequenceOrRecordExpression = not implicitYieldEnabled
 
         match comp with
-        | SynExpr.New _ ->
```
Comment (edgarfgp) [inline src/Compiler/Checking/Expressions/CheckSequenceExpressions.fs:454]: To help reviewers: This specific change was removed in https://github.com/dotnet/fsharp/pull/17388/files but added back in previous refactoring PR by mistake.

**Review by KevinRansom — DISMISSED** [review]
These test cases seem to verify that the compiler compiles the source code without producing errors, could we also verify that the built type has the interface.  Adding a main with a cast and executing is probably sufficient.

I'm confident it does, but a test that confirms it might be useful.

**Comment by edgarfgp** [comment]
> These test cases seem to verify that the compiler compiles the source code without producing errors, could we also verify that the built type has the interface. Adding a main with a cast and executing is probably sufficient.
> 
> I'm confident it does, but a test that confirms it might be useful.

Done 👍🏻

**Review by psfinaki — APPROVED** [review]

**Review by vzarytovskii — APPROVED** [review]

**Pull request merged**